includes/model.php

```diff
@@ -654,6 +654,9 @@ public static function get_image_by_url( string $url = '' ): ?int {
 
 		ISC_Log::log( 'enter get_image_by_url() to look for URL ' . $url );
 
+		$original_url = $url;
+		$url          = apply_filters( 'isc_filter_url_pre_get_image_by_url', $url );
```

Comment:
**Suggestion:** It's a good practice to validate the input before using it. Consider validating `$url` before applying the filter at line 658. [best practice]
```suggestion
  $original_url = $url;
  if (filter_var($url, FILTER_VALIDATE_URL)) {
      $url = apply_filters( 'isc_filter_url_pre_get_image_by_url', $url );
  }
```